### PR TITLE
Pull Request for Issue2393: Creating generic scan config view for BioXAS

### PR DESCRIPTION
**Pull request by helfrij**
Created:
- BioXASGenericStepScanConfigurationView, a new view for generic step scans on BioXAS.
- BioXASGenericStepScanConfigurationAxesView, a new view for configuration axes. This view defaults to showing information for the first two axes.
- BioXASGenericStepScanConfigurationAxisView, a new view for displaying information about a single scan axis.
- BioXASScanAxisRegionView, a new view that displays region information in either Absolute or Relative mode.

Modified:
- Fixed bug in AMGenericStepScanConfigurationDetectorsView where the application would crash if a detector was selected when the configuration for the view wasn't set. Now, the detectors are disabled if that's the case.
- Added some signals to AMScanConfiguration to provide additional information about axis controls, when they are changed/added/removed. Modified the AMGenericStepScanConfiguration to emit these signals as appropriate.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you review?

File: source/acquaman/AMGenericStepScanConfiguration.cpp [hunk 1]
```diff
@@ -197,12 +200,16 @@ void AMGenericStepScanConfiguration::setControl(int axisId, AMControlInfo newInf
 		connect(scanAxisAt(1)->regionAt(0), SIGNAL(regionStepChanged(AMNumber)), this, SLOT(computeTotalTime()));
 		connect(scanAxisAt(1)->regionAt(0), SIGNAL(regionEndChanged(AMNumber)), this, SLOT(computeTotalTime()));
 		connect(scanAxisAt(1)->regionAt(0), SIGNAL(regionTimeChanged(AMNumber)), this, SLOT(computeTotalTime()));
+
+		emit axisControlInfoAdded();
+		emit axisControlInfoChanged();
 	}
```
Comment (XuetaoLiu-CLS) [inline source/acquaman/AMGenericStepScanConfiguration.cpp]: I saw potential of code generalization here...
codes to add / replace axis 0 and 1 are very alike

**Comment by XuetaoLiu-CLS** [comment]
Looks good. Nice job :100: 

Time Estimate: 20 minutes

**Comment by dretrex** [comment]
Looks great!
This can/should likely become the base class of other views of this type.  Awesome stuff.
Time Estimate: 10 minutes

**Comment by dretrex** [comment]
There are some merge conflicts to deal with, but other than that it is good to go.

**Comment by helfrij** [comment]
Merge conflicts should be resolved. Thanks @XuetaoLiu-CLS @dretrex!

**Pull request merged**